compiler/src/main/scala/org/scalajs/nscplugin/ExplicitLocalJS.scala

```diff
@@ -394,6 +394,21 @@ abstract class ExplicitLocalJS[G <: Global with Singleton](val global: G)
       }
     }
 
+    /** Wraps with the contextual super JS class value for super calls. */
+    private def wrapWithContextualSuperJSClassValue(superClass: Symbol)(
+        tree: Tree): Tree = {
+      /* #4801 We need to interpret the superClass type as seen from the
+       * current class' thisType.
+       */
+      val jsClassTypeInSuperClass = superClass.tpe_*
+      val jsClassTypeAsSeenFromThis =
+        jsClassTypeInSuperClass.asSeenFrom(currentClass.thisType, superClass)
```

Comment:
Just to check my understanding: This "translates" the type from `OuterNativeClass_Issue4402#InnerClass` to the path dependent type `outer.InnerClass`.

Reply:
To be more precise, it translates the type from `OuterNativeClass_Issue4402.this.InnerClass` (which is path-dependent on the outer `this` value) into the path-dependent type `outer.InnerClass`. I have expanded the comment above to make it clearer.